compiler/rustc_error_codes/src/error_codes/E0311.md

```diff
@@ -0,0 +1,49 @@
+E0311 occurs when there is insufficient information for the rust compiler to
+prove that some time has a long enough lifetime.
```

Comment:
Thank you for the contributions! This explanation could be a bit confusing, I don't know how it should be reworded into, cc @compiler-errors any ideas?

Reply:
Good point -- it would be great to have a better description here. I'm still pretty new to rust, so this was my initial attempt at reverse engineering why the fix worked.